models/orogen/rock_gazebo.rb

```diff
@@ -78,40 +88,54 @@ def self.common_dynamic_link_export(context, name, port_name = name, options)
     # at the same time.
     dynamic_service CommonModels::Devices::Gazebo::Model, as: 'submodel_export' do
         name = self.name
-        _, frame_basename = OroGen::RockGazebo::ModelTask.common_dynamic_link_export(
-            self, self.name, options.merge!(port_name: "#{name}_pose_samples"))
+        OroGen::RockGazebo::ModelTask.common_dynamic_link_export(
+            self, name, options.merge!(port_name: "#{name}_pose_samples")
+        )
         driver_for CommonModels::Devices::Gazebo::Model,
-            'joints_cmd'   => "#{name}_joints_cmd",
-            'joints_status'   => "#{name}_joints_samples"
+                   'joints_cmd' => "#{name}_joints_cmd",
+                   'joints_status' => "#{name}_joints_samples"
     end
 
     transformer do
         transform_output 'pose_samples', 'model' => 'world'
     end
 
+    def period_to_time(period)
+        if period
+            period_us = (period * 1_000_000).round
+            Time.at(period_us / 1_000_000, period_us % 1_000_000, :usec)
+        else
+            Time.at(0)
+        end
+    end
```

Comment:
Why not just:

```ruby
def period_to_time(period)
    period_us = (period * 1_000_000).round
    Time.at(period_us / 1_000_000, period_us % 1_000_000, :usec)
end
```

?

Reply:
I guess you meant

~~~
    period_us = ((period || 0) * 1_000_000).round
    Time.at(period_us / 1_000_000, period_us % 1_000_000, :usec)
~~~

?

Reply:
Yep